src/interpreter/types/symbolTable.ts

```diff
@@ -0,0 +1,28 @@
+export interface SymbolTable {
+  head: SymbolTableFrame;
+  tail: SymbolTable | null;
+  parent: FunctionSymbolTableEntry | null;
+}
+
+export type SymbolTableFrame = Record<string, SymbolTableEntry>;
+export type SymbolTableEntry =
+  | FunctionSymbolTableEntry
+  | VariableSymbolTableEntry;
+export type SymbolTableEntryScope = 'Block' | 'Function' | 'Global';
+
+export interface BaseSymbolTableEntry {
+  name: string;
+  nameType: string;
```

Comment:
Minor nit: can we shift this to be the first property in the object? I think it would be a good convention to establish - that the type that we discriminate unions on is the very first type listed.